projects/laji/src/app/+observation/observation.facade.ts

```diff
@@ -126,17 +121,26 @@ export class ObservationFacade {
             query[key] =  loggedIn ? this.userService.getToken() : undefined;
           }
         });
+
+        const newQuery = Util.clone(query);
+
         const hash = JSON.stringify(warehouseQuery);
-        if (this.hashCache['query'] === hash) {
+        if (this.queryHash === hash) {
+          this.updateState({..._state, newQuery, newQueryHasChanges: false});
           return;
         }
-        this.hashCache['query'] = hash;
+        this.queryHash = hash;
 
-        this.updateState({..._state, query, loadingUnits: true, loadingTaxa: true});
+        this.updateState({..._state, query, newQuery, newQueryHasChanges: false, loadingUnits: true, loadingTaxa: true});
       })
     );
   }
 
+  updateNewQuery(warehouseQuery: WarehouseQueryInterface) {
```

Comment:
The naming of these could also be more explicit, took me a while to understand the difference between these:

`updateQuery` -> `updateActiveQuery`
`updateNewQuery` -> `updateTmpQuery`